src/util/cert/libcrypto/cert.c

```diff
@@ -176,31 +177,116 @@ errno_t sss_cert_pem_to_der(TALLOC_CTX *mem_ctx, const char *pem,
 #define IDENTIFIER_NISTP384 "nistp384"
 #define IDENTIFIER_NISTP521 "nistp521"
 
+static int sss_ec_get_key(BN_CTX *bn_ctx, EVP_PKEY *cert_pub_key,
```

Comment:
Won't `bn_ctx` generate "unused parameter" warning with old openssl?

Reply:
`cert_pub_key` can be a pointer to 'const' in case of openssl 3.0.0+

Reply:
Yes, an "unused parameter" warning might be seen but currently we ignore this warning in other places as well.

I change the code so that `cert_pub_key` can be `const` for all versions.